flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CompletedCheckpoint.java

```diff
@@ -368,6 +373,34 @@ public void discard() throws Exception {
             }
         }
 
+        @Override
+        public void discard() throws Exception {
+            discard(null);
+        }
+
+        private void discardOperatorStates(Executor ioExecutor) throws Exception {
```

Comment:
The logic is similar with PendingCheckpoint, right ?
Could you unify and put the logic into one method ?

Reply:
yes, the logic is the same. Added default method discardOperatorStates in Checkpoint interface to unify.